uaclient/api/data_types.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict, List, Union  # noqa: F401
+
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.version import get_version
+
+
+class AdditionalInfo:
+    meta = {}  # type: Dict[str, str]
+    warnings = []  # type: List[ErrorWarningObject]
+
+
+class ErrorWarningObject(DataObject):
+    fields = [
+        Field("title", StringDataValue),
+        Field("code", StringDataValue),
+        Field("meta", DataObject),
+    ]
+
+    def __init__(
+        self, *, title: str, code: str, meta: Union[DataObject, dict]
+    ):
+        self.title = title
+        self.code = code
+        self.meta = meta
+
+
+class APIData(DataObject):
+    fields = [
+        Field("type", StringDataValue),
+        Field("attributes", DataObject),
+        Field("meta", DataObject),
+    ]
+
+    def __init__(self, *, type: str, attributes: DataObject, meta: DataObject):
+        self.type = type
+        self.attributes = attributes
+        self.meta = meta
+
+
+class APIResponse(DataObject):
+    fields = [
+        Field("_schema_version", StringDataValue),
+        Field("result", StringDataValue),
+        Field("version", StringDataValue),
```

Comment:
Sorry I'm just realizing this is possibly my last chance for bikeshedding and trying to get it all in. I'm okay if you deny this.

```suggestion
        Field("_schema_version", StringDataValue),
        Field("result", StringDataValue),
        Field("version", StringDataValue),
        Field("endpoint_version", StringDataValue)
```
`_schema_version` should be hardcoded to "v1" as this is should be changed to represent the version of the overall API schema.
`endpoint_version` is where we'd put the version string of the endpoint

Reply:
hhhm, not sure if I understood this - would not the schema be tied to each endpoint? How will we manage an overall API schema?

Reply:
we talked out of band and decided this is good as is :+1: